src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,829 @@
+                }
+
+                await UpdateExpressionInOriginalFunctionAsync(originalDocument, expression, containingMethod,
+                    parameterName, editor, allOccurrences, cancellationToken).ConfigureAwait(false);
+                var semanticModel = await originalDocument.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var parameterType = semanticModel.GetTypeInfo(expression, cancellationToken).ConvertedType ?? semanticModel.Compilation.ObjectType;
+                var refKind = syntaxFacts.GetRefKindOfArgument(expression);
+                var parameter = generator.ParameterDeclaration(name: parameterName, type: generator.TypeExpression(parameterType), refKind: refKind);
+                editor.InsertParameter(containingMethod, insertionIndex, parameter);
+            }
+
+            return modifiedSolution.WithDocumentSyntaxRoot(originalDocument.Id, editor.GetChangedRoot());
+
+            // Adds an argument which is an invocation of the newly created method to the callsites
+            // of the method invocations where a parameter was added.
+            // Example:
+            // public void M(int x, int y)
+            // {
+            //     int f = x * y; // highlight this expression
+            // }
+            // 
+            // public void M1()
+            // {
+            //     M(5, 6);
+            // }
+            //
+            // ---------------------------------------------------->
+            // 
+            // public int M_f(int x, int y)
+            // {
+            //     return x * y;
+            // }
+            // 
+            // public void M(int x, int y)
+            // {
+            //     int f = x * y;
+            // }
+            //
+            // public void M1()
+            // {
+            //     M(5, 6, M_f(5, 6)); // This is the generated invocation which is a new argument at the call site
+            // }
+            static TInvocationExpressionSyntax GenerateNewInvocationExpressionForTrampoline(ISyntaxFactsService syntaxFacts,
+                SyntaxEditor editor, SyntaxNode currentInvocation, TInvocationExpressionSyntax invocationExpression,
+                string newMethodIdentifier, int insertionIndex)
+            {
+                var invocationArguments = syntaxFacts.GetArgumentsOfInvocationExpression(currentInvocation);
+                var methodName = editor.Generator.IdentifierName(newMethodIdentifier);
+                var expressionFromInvocation = syntaxFacts.GetExpressionOfInvocationExpression(invocationExpression);
+                var newMethodInvocation = editor.Generator.InvocationExpression(methodName, invocationArguments);
+                var allArguments = invocationArguments.Insert(insertionIndex, newMethodInvocation);
+                return (TInvocationExpressionSyntax)editor.Generator.InvocationExpression(expressionFromInvocation, allArguments);
```

Comment:
i would not do this.

first, you are losing trivia (please add test).  
second, you are regenerating far more than you need do.  alll you are doing is *adding* an arg to an *existing* invocation.  you dont' want to replace the whole invocation.  you just need to do that single addition.

Reply:
How would trivia be lost? And how would I be able to add an arg to an exisiting invocation? Would I just need to find the argument prior to the insertion index and do editor.InsertAfter?

Reply:
nevermind you explained it, also changed already